lightning/src/ln/channelmanager.rs

```diff
@@ -3103,37 +2956,13 @@ where
 			}
 		}
 
-		let next_hop = match onion_utils::decode_next_payment_hop(
-			shared_secret, &msg.onion_routing_packet.hop_data[..], msg.onion_routing_packet.hmac,
-			msg.payment_hash, &self.node_signer
-		) {
-			Ok(res) => res,
-			Err(onion_utils::OnionDecodeErr::Malformed { err_msg, err_code }) => {
-				return_malformed_err!(err_msg, err_code);
-			},
-			Err(onion_utils::OnionDecodeErr::Relay { err_msg, err_code }) => {
-				return_err!(err_msg, err_code, &[0; 0]);
-			},
-		};
-		let (outgoing_scid, outgoing_amt_msat, outgoing_cltv_value, next_packet_pk_opt) = match next_hop {
-			onion_utils::Hop::Forward {
-				next_hop_data: msgs::InboundOnionPayload::Forward {
-					short_channel_id, amt_to_forward, outgoing_cltv_value
-				}, ..
-			} => {
-				let next_packet_pk = onion_utils::next_hop_pubkey(&self.secp_ctx,
-					msg.onion_routing_packet.public_key.unwrap(), &shared_secret);
-				(short_channel_id, amt_to_forward, outgoing_cltv_value, Some(next_packet_pk))
-			},
-			// We'll do receive checks in [`Self::construct_pending_htlc_info`] so we have access to the
-			// inbound channel's state.
-			onion_utils::Hop::Receive { .. } => return Ok((next_hop, shared_secret, None)),
-			onion_utils::Hop::Forward { next_hop_data: msgs::InboundOnionPayload::Receive { .. }, .. } |
-				onion_utils::Hop::Forward { next_hop_data: msgs::InboundOnionPayload::BlindedReceive { .. }, .. } =>
-			{
-				return_err!("Final Node OnionHopData provided for us as an intermediary node", 0x4000 | 22, &[0; 0]);
-			}
-		};
+		// it is a receive, so no need for outbound checks
+		if let None = next_packet_details_opt {
```

Comment:
nit: this is more idiomatic, same applies below 
```suggestion
		if next_packet_details_opt.is_none() {
```